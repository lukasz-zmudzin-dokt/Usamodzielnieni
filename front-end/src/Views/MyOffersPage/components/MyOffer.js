import React from "react";
import {Accordion, Card} from "react-bootstrap";
import InterestedPerson from "./InterestedPerson";
import MyOffersLegend from "./MyOffersLegend";
import { getInterestedPeople } from "../functions/getInterestedPeople";
import "../style.css";

const MyOffer = ({ offer, token, component})  => {
    return (
        <Card className="border">
            <Accordion.Toggle as={Card.Header} eventKey={offer.id} onClick={e => (getInterestedPeople(token, offer.id)).then(r => component.setState({answers: r}))}>
                {offer.offer_name}
            </Accordion.Toggle>
            <Accordion.Collapse eventKey={offer.id}>
                <Card.Body>
                    <MyOffersLegend answers={component.state.answers} />
                    {component.state.answers.map((value) => {
                        return (
                            <InterestedPerson person={value} key={value.id}/>
                        )
                    })}
                </Card.Body>
            </Accordion.Collapse>
        </Card>
    );
};

export default MyOffer;