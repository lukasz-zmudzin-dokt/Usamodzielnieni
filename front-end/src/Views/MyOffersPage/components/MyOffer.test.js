import React from "react";
import { render, waitForElement, fireEvent } from "@testing-library/react";
import MyOffer from "./MyOffer";
import { MemoryRouter } from "react-router-dom";

describe("MyOffers", () => {
    let fetchCheck;
    let testOffer = {
        category: "IT",
        company_address: "adress1",
        company_name: "company1",
        description: "description123",
        expiration_date: "2020-05-28",
        id: "47991e86-4b42-4507-b154-1548bf8a3bd3",
        offer_name: "offer2",
        type: "StaÅ¼",
        voivodeship: "mazowieckie"
    };
    let apiPeople = [
        {
            cv_url: "/media/blank_test_cv",
            date_posted: "2020-04-06T01:34:27.899000+02:00",
            email: "standard1@standard1.com",
            first_name: "standard1",
            job_offer: "47991e86-4b42-4507-b154-1548bf8a3bd3",
            last_name: "standard1",
            user_id: "b582b042-d6d8-4e57-9447-564a6748b4f7"
        }
    ];

    beforeAll(() => {
        global.fetch = jest.fn().mockImplementation((input, init) => {
            return new Promise((resolve, reject) => {
                switch (fetchCheck) {
                    case "fail":
                        resolve({ status: 500 });
                        break;
                    case "empty":
                        resolve({ status: 200, json: () =>Promise.resolve([]) });
                        break;
                    case "normal":
                        resolve({ status: 200, json: () =>Promise.resolve(apiPeople) });
                        break;
                    default:
                        reject([]);
                }
            });
        });
    });

    beforeEach(() => {
        fetchCheck = "normal";
        jest.clearAllMocks();
    });

    it('should match snapshot', async () => {
        const { container, getByText } = render (
            <MemoryRouter>
                <MyOffer offer={testOffer} />
            </MemoryRouter>
        );
        await waitForElement(() => getByText("offer2"));

        expect(container).toMatchSnapshot();

    });

});


