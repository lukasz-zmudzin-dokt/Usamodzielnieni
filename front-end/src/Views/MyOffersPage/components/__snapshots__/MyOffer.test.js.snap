// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyOffers should match snapshot 1`] = `
<div>
  <div
    class="border-left-0 border-right-0 border-bottom-0 card"
  >
    <div
      class="card-header"
    >
      offer2
    </div>
    <div
      class="collapse"
    >
      <div
        class="p-0 card-body"
      >
        <div
          class="fade mb-0 alert alert-info show"
          role="alert"
        >
          Brak zgłoszeń.
        </div>
        <div
          class="list-group"
        >
          <div
            class="list-group-item"
          >
            <div
              class="justify-content-center row"
            >
              <a
                href="/jobOffers/47991e86-4b42-4507-b154-1548bf8a3bd3"
              >
                <button
                  class="btn btn-primary"
                  type="button"
                >
                  Pokaż ofertę
                </button>
              </a>
              <a
                href="/offerForm/47991e86-4b42-4507-b154-1548bf8a3bd3"
              >
                <button
                  class="ml-3 btn btn-primary"
                  type="button"
                >
                  Edytuj ofertę
                </button>
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
